services/appmetrics/api.go

```diff
@@ -0,0 +1,46 @@
+package appmetrics
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/appmetrics"
+)
+
+func NewAPI (db *appmetrics.Database, metricsBufferedChan chan appmetrics.AppMetric) *API {
+	return &API{db: db, metricsBufferedChan: metricsBufferedChan}
+}
+
+type API struct {
+	db *appmetrics.Database
+	metricsBufferedChan chan appmetrics.AppMetric
+}
+
+func (api *API) ValidateAppMetrics(ctx context.Context, appMetrics []appmetrics.AppMetric) error {
+	log.Info("[AppMetricsAPI::ValidateAppMetrics]")
+	return api.db.ValidateAppMetrics(appMetrics)
+}
+
+func (api *API) SaveAppMetrics(ctx context.Context, appMetrics []appmetrics.AppMetric) error {
+	log.Info("[AppMetricsAPI::SaveAppMetrics]")
+	chanFull := len(api.metricsBufferedChan) == cap(api.metricsBufferedChan)
```

Comment:
Honestly I would not bother having a buffer here, I doubt it's going to impact performance and it's probably just added complexity, this will be run on a separate thread so won't impact the UI thread or the JS thread.

Reply:
So we just write to DB each time an event occurs? 

We don't really know the frequency of these events, but we know that this frequency will grow over time.

Reply:
I'd say so, unless write are multiple times per second (4/5) I don't think it will have an impact performance wise, the RPC call in the UI thread and handling of serialization is likely more expensive, so any batching is best done on the client if performance are a concern (I'd have to time it, but we know UI/Cljoure is much slower than go code, so you'll be likely hitting your bottleneck on the client before having issue with db writes)

Reply:
The way we have built this, might lead to 2-5 calls at once, followed by a few seconds of inactivity. 

Since we use this in conjunction with Re-frame event loop, we can easily be in a situation where we decide to capture 3 events, which get fired almost immediately. I'm trying to think of an example. 

One of the reasons why I pushed batching to Go side is because it can be re-used by Desktop team.

Reply:
Probably you want to batch those calls in status-react if you are making them, I would just "debounce" them, and have this endpoint rather receive an array and insert as soon as it receives them.
We have seen this before, basically the bottleneck it's status-react/the bridge, if you make too many calls, status-go will never be hit hard enough (provided that the workload is small on status-go of course, like in this case), but the bridge will choke